plugins/pulp_rpm/plugins/db/models.py

```diff
@@ -480,6 +486,112 @@ def rpm_search_dicts(self):
+        :rtype: bool
+        """
+        if len(existing_packages) == len(new_packages):
+            existing_packages.sort(key=itemgetter('filename'))
+            new_packages.sort(key=itemgetter('filename'))
+            return existing_packages == new_packages
+        return False
+
+    def merge_errata(self, other):
+        """
+        Merge two errata with the same errata_id.
+
+        There are two parts:
+        - merging of the pkglists in case of the erratum with the same id in different repositories
+        - overwriting the erratum metadata based on the `updated` field
+
+        NOTE: The second part should be eliminated after we change the way erratum is stored in the
+        MongoDB.
+
+        :param other: The erratum we are combining with this one
+        :type  other: pulp_rpm.plugins.db.models.Errata
+        """
+        self.merge_pkglists(other)
+        if self.update_needed(other):
+            for field_name in self.mutable_erratum_fields:
+                setattr(self, field_name, getattr(other, field_name))
+
+    def update_needed(self, other):
+        """
+        Decide based on the `updated` field if the update of the existing erratum is needed.
+
+        The `updated` field is just a string in the MongoDB, so there is no strict format for this
+        date-time field. If we are not able to parse the `updated` field either in existing
+        erratum or in the new erratum, the metadata of existing erratum won't be updated.
+
+        :param other: potentially a newer version of the erratum
+        :type  other: pulp_rpm.plugins.db.models.Errata
+
+        :return: True if the other erratum is newer than the existing one
+        :rtype:  bool
+        """
+        err_msg = _('Fail to update the %(which)s erratum %(id)s: '
+                    'Unable to parse the `updated` field')
+        existing_err_msg = err_msg % {'which': 'existing', 'id': self.errata_id}
+        other_err_msg = err_msg % {'which': 'uploaded', 'id': self.errata_id}
+        existing_updated_dt = util.errata_format_to_datetime(self.updated, msg=existing_err_msg)
+        new_updated_dt = util.errata_format_to_datetime(other.updated, msg=other_err_msg)
+        return new_updated_dt > existing_updated_dt
+
+    def merge_pkglists(self, other):
+        """
+        Merge pkglists of the two errata.
+
+         - add _pulp_repo_id to old collection if packages are the same
+         - update existing collection if the other collection is newer and from the same
+           repository
+         - otherwise add a new collection
+
+        :param other: The erratum we are combining with the existing one
+        :type  other: pulp_rpm.plugins.db.models.Errata
+
+        """
+        existing_pkglist_map = {}
+        for idx, p in enumerate(self.pkglist):
+            package_name = p['name']
+            package_repo_id = p.get('_pulp_repo_id')
+            existing_pkglist_map[package_name, package_repo_id] = idx
```

Comment:
I see now it's a tuple. For clarity can the creation of the tuple key be moved to its own line like:

```
pkglist_key = (package_name, package_repo_id)
existing_pkglist_map[pkglist_key] = idx
```